lib/beaker/host/unix/exec.rb

```diff
@@ -85,23 +85,24 @@ def reboot(wait_time=10, max_connection_tries=9, uptime_retries=18)
       @logger.debug("Original Boot Time: #{original_boot_time}")
       @logger.debug("Current Boot Time: #{current_boot_time}")
 
+      # This actually may fail a few times before succeeding because virtual
+      # machines can skew backwards in time so we need to let the retry happen.
       unless current_boot_time > original_boot_time
         raise Beaker::Host::RebootFailure, "Boot time did not reset. Reboot appears to have failed."
       end
-    rescue Beaker::Host::RebootFailure => e
+    rescue ArgumentError => e
+      raise Beaker::Host::RebootFailure, "Unable to parse time: #{e.message}"
+    rescue => e
+      # Yes, this isn't best practice but rebooting is weird and we really want
+      # to make sure we retry if something went wrong (no matter what it was,
+      # including hosts going back in time).
       attempts += 1
```

Comment:
are we instructing beaker to reboot the system again, in case we can't tell it was rebooted?

Reply:
@igalic  No, the system will try to reboot multiple times if there is some issue getting the original information from the system.

The second loop set will not try to reboot, it will just try to access the system until the count limit is reached.